server/sonar-process-monitor/src/test/java/org/sonar/application/cluster/AppStateClusterImplTest.java

```diff
@@ -40,14 +44,19 @@
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
 import static org.sonar.application.cluster.AppStateClusterImpl.OPERATIONAL_PROCESSES;
+import static org.sonar.application.cluster.AppStateClusterImpl.SONARQUBE_VERSION;
+import static org.sonar.application.cluster.HazelcastTestHelper.createHazelcastClient;
+import static org.sonar.application.cluster.HazelcastTestHelper.createHazelcastNode;
+import static org.sonar.application.cluster.HazelcastTestHelper.getFreePortOn;
+import static org.sonar.application.config.SonarQubeVersionHelper.getSonarqubeVersion;
 
 public class AppStateClusterImplTest {
 
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
   @Rule
-  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(10));
+  public TestRule safeGuard = new DisableOnDebug(Timeout.seconds(20));
```

Comment:
The "m" is missing :-) Duration should be ~10ms !
Let's try the following approach:
* extract AppState#registerSQVersion(String). Throws ISE if version different than the first registered version.
* App calls appState.registerSQVersion(loadSQVersion())
* AppStateClusterImplTest does not need to start a Hazelcast cluster. It calls twice registerSQVersion().